python/setup.py

```diff
@@ -0,0 +1,43 @@
+import codecs
+import glob
+import os
+
+import setuptools
+
+
+def read(*parts):
+    """Read parts to use a e.g. long_description"""
+
+    here = os.path.abspath(os.path.dirname(__file__))
+
+    # intentionally *not* adding an encoding option to open, See:
+    #   https://github.com/pypa/virtualenv/issues/201#issuecomment-3145690
+    with codecs.open(os.path.join(here, *parts), "r") as pfp:
+        return pfp.read()
+
+
+setuptools.setup(
+    name="xnvme",
+    version="0.3.0",
+    author="Simon A. F. Lund",
+    author_email="os@safl.dk",
+    description="xNVMe Cython and ctypes language-bindings for Python",
+    long_description=read("README.rst"),
+    url="https://github.com/OpenMPDK/xNVMe",
+    project_urls={
+        "Bug Tracker": "https://github.com/OpenMPDK/xNVMe/issues",
+    },
+    license="Apache License, Version 2.0",
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: Apache Software License",
+        "Operating System :: OS Independent",
+    ],
+    data_files=[
+        ("bin", glob.glob("bin/*")),
+    ],
+    package_dir={"": "./"},
+    packages=setuptools.find_packages(where="./"),
+    python_requires=">=3.6",
+    install_requires="pytest",
```

Comment:
Install wouldn't require `pytest`. Either use `tests_require` or now that you have a `python/pyproject.toml`, put it there instead.

Reply:
I have gone with the approach of installing the testcases, this allows the ease-of-use of running the pytest via the package-discovery. This is why pytest is added as a runtime requirement.

Reply:
I still think we should not have it here, and put it in either `tests_require` or `pyproject.toml`, as it's not part of normal runtime usage.

Also, isn't it supposed to be a list? `["pytest"]`?

Reply:
The reason not to do it, would be because it's not a runtime dependency. I think it would be exceedingly rare somebody else is running our test suite. And as soon as we merge my Cython PR, the tests would no longer succeed without proper configuration.

But why not put it under the test requirements, if it's for testing?

Reply:
Adding to that, this is why ``python3 -m pytest --pyargs xnvme`` is added as a bold note in the docs right after installation.

Reply:
Also, this does not dictate that all tests are installed with the package, just the ones useful for the use-case described above.

Reply:
How about a different approach? Instead of implementing these as a pytest, why not just a simple function?

```py
def self_test():
     # Self-test for the dynamically loaded library

     # The 'capi' is loaded upon module access, this checks that it succeeded
     assert capi.is_loaded, "C API isn't loaded"

     # This verifies that the loaded library actually is the xNVMe library, by checking for
     # the existance of a xNVMe specific funtion.
     assert capi.xnvme_enumerate, "xnvme_enumerate is not present in library. Library isn't loaded correctly"

     # Verifies that at at least one function is callable
     assert capi.xnvme_libconf_pr(capi.XNVME_PR_DEF), "Failed to call function 'xnvme_libconf_pr in xNVMe. Library isn't loaded correctly"

     print('xNVMe ctypes bindings loaded correctly!')
```

Then in the docs: `python3 -c 'import xnvme.ctypes_bindings as xnvme; xnvme.self_test()'`

Reply:
That is also an approach, the down-side is that then the tests need to added in multiple places and not behaving as nicely as ``pytest``.

I do like the messages on the asserts, so I have added that to the tests.

I don't see any reason not to use ``pytest`` for this, the only counter-argument is that it is "unusual" but it is no more unusual than ``pytest`` describing the use-case in their documentation and providing test-collection for it. So, its a well-documented and supported feature of ``pytest``.

Should the addition of ``pytest`` cause us grief, then it is no worse than we can remove it and replace it with something like the approach above. However, for now I see no reason to, it only makes things simpler/easier to use.

Reply:
> the only counter-argument is that it is "unusual"

The other argument is that this solution is short-lived. As soon as we merge my Cython-to-Python bindings, it won't work anymore.

> Should the addition of pytest cause us grief, then it is no worse than we can remove it and replace it with something like the approach above.

If you want to, then keep and let's see.